python_modules/dagster/dagster/_cli/api.py

```diff
@@ -316,9 +318,21 @@ def verify_step(instance, pipeline_run, retry_state, step_keys_to_execute):
     ),
 )
 @click.argument("input_json", type=click.STRING, envvar="DAGSTER_EXECUTE_STEP_ARGS")
-def execute_step_command(input_json):
+@click.option(
+    "commpressed",
+    "--compressed",
+    is_flag=True,
+    default=False,
+    envvar="DAGSTER_CLI_API_EXECUTE_STEP_COMPRESSED",
+)
+def execute_step_command(input_json, commpressed):
     with capture_interrupts():
 
+        print("HERE", commpressed, input_json)
+
+        if commpressed:
+            input_json = zlib.decompress(base64.b64decode(input_json.encode())).decode()
+
```

Comment:
I think it would be clearer as two different inputs instead of one that changes based on another arg - adding an `xor` check isn't too bad

Reply:
done